util/src/main/java/org/folio/reservoir/util/readstream/Mapper.java

```diff
@@ -0,0 +1,9 @@
+package org.folio.reservoir.util.readstream;
+
+public interface Mapper<T, V> {
+
+  public void put(T item);
+
+  public V get(boolean ended);
```

Comment:
I think I slightly prefer a separate end method get is "poll". And put is "push". So it seems natual to have an end "push" and not mix it with the polling get.

```suggestion
  public V get();
  
  public void end();
```